autogpt/llm/providers/openai.py

```diff
@@ -228,7 +229,7 @@ def create_chat_completion(
         OpenAIObject: The ChatCompletion response from OpenAI
 
     """
-    completion: OpenAIObject = openai.ChatCompletion.create(
```

Comment:
I have concerns that we are directly replacing the provider here without an architecture to support this.

We do a lot more than call the completion such as tracking costs.

Reply:
hmm .. given community effort to create similar variants with other providers (e.g. https://github.com/LostAbaddon/AutoClaude) 

i'd be curious to see how we could do that in a 'safe' manner?

Reply:
maybe if there's an 'other' provider class - we could potentially use something like 'litellm' to handle the abstraction calling there 

happy to work with @ishaan-jaff on enabling this.

Reply:
I could see that working pretty well. @collijk ?